src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc

```diff
@@ -0,0 +1,1505 @@
+runtimeOnly 'org.apache.logging.log4j:log4j-jpa'
+----
+====
+
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-jpa_org-apache-logging-log4j-core-appender-db-jpa-JpaAppender[{open-book} Plugin reference for `JPA`]
+
+[#JpaAppender-examples]
+=== Configuration examples
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="error">
+  <Appenders>
+    <JPA name="databaseAppender" persistenceUnitName="loggingPersistenceUnit"
+         entityClassName="com.example.logging.JpaLogEntity" />
+  </Appenders>
+  <Loggers>
+    <Root level="warn">
+      <AppenderRef ref="databaseAppender"/>
+    </Root>
+  </Loggers>
+</Configuration>
+----
+
+[#NoSqlAppender]
+== NoSQL Appender
+
+The NoSQL Appender writes log events to a NoSQL database using an internal lightweight provider interface.
+It supports the following configuration options
+
+[#NoSqlAppender-attributes]
+.NoSQL Appender configuration attributes
+[cols="1m,1,1,5"]
+|===
+| Attribute | Type | Default value | Description
+
+4+h| Required
+
+| [[NoSqlAppender-attr-name]]name
+| `String`
+|
+| The name of the Appender.
+
+4+h| Optional
+
+| [[NoSqlAppender-attr-bufferSize]]bufferSize
+| `int`
+| `0`
+|
+The number of log messages to batch before writing to the database.
+If `0`, batching is disabled.
+
+| [[NoSqlAppender-attr-ignoreExceptions]]ignoreExceptions
+| `boolean`
+| `true`
+| If `false`, logging exception will be forwarded to the caller of the logging statement.
+Otherwise, they will be ignored.
+
+|===
+
+[#NoSqlAppender-elements]
+.NoSQL Appender nested elements
+[cols="1m,1,4"]
+|===
+| Type | Multiplicity | Description
+
+| [[NoSqlAppender-element-Filter]]xref:manual/filters.adoc[`Filter`]
+| zero or one
+|
+Allows filtering log events just before they are formatted and sent.
+
+See also xref:manual/filters.adoc#appender-stage[appender filtering stage].
+
+| [[NoSqlAppender-element-KeyValuePair]]xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-core_org-apache-logging-log4j-core-util-KeyValuePair[`KeyValuePair`]
+| Zero or more
+a| Adds a simple key/value field to the NoSQL object.
+
+The `value` attribute of the pair supports
+xref:manual/configuration.adoc#lazy-property-substitution[runtime property substitution]
+using the
+xref:manual/lookups.adoc#event-context[current event as context].
+
+| [[NoSqlAppender-element-Layout]]<<NoSqlAppender-formatting,`Layout`>>
+| zero or one
+|
+An optional
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/Layout.html[`Layout<? extends MapMessage>`]
+implementation that formats a log event as
+xref:manual/messages.adoc#MapMessage[`MapMessage`].
+
+See <<NoSqlAppender-formatting>> for more details.
+|===
+
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-core_org-apache-logging-log4j-core-appender-nosql-NoSqlAppender[{open-book} Plugin reference for `NoSQL`]
+
+[#NoSqlAppender-formatting]
+=== Formatting
+
+This appender transforms log events into JSON-like structures in two ways:
```

Comment:
I am confused about this `JSON-like structure`. Is it JSON or not? If not, how is it _alike_?

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/11fa9ec4f4d2b98aa6e901f9ba0de0d5ef28deb6